app/controllers/posts_controller.rb

```diff
@@ -7,4 +10,32 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def create
+    @post = current_user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def find_user
+    @user = User.find(params[:user_id])
+  end
+
+  def find_post
+    @post = Post.find(params[:id])
+  end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
+
+  def new
+    @post = Post.new
+  end
```

Comment:
- [ ] Great work adding the `new` method as instructed by the previous reviewer. However, This method should not be a private method. It seems you have added it at the bottom of the files among the private methods. Kindly remove it from here and place it at the top. Remember this is a method that is used by both the routes and the views. So it can not be private to this class. ⬇️ 

![image](https://github.com/shm04/blogApp/assets/32361963/4598bdb3-46ab-49aa-8feb-703bee15e5f4)